extensions/wikia/AdEngine3/src/slots.js

```diff
@@ -399,5 +399,22 @@ export default {
 				'floor_adhesion',
 				{ distanceFromTop: utils.getViewportHeight() },
 		);
+
+		let porvataClosedActive = false;
+
+		slotService.on('floor_adhesion', AdSlot.STATUS_SUCCESS, () => {
+			porvataClosedActive = true;
+
+			eventService.on(events.VIDEO_AD_IMPRESSION, () => {
+				if (porvataClosedActive) {
+					porvataClosedActive = false;
+					slotService.disable('floor_adhesion', 'closed-by-porvata');
+				}
+			});
+		});
+
+		slotService.on('floor_adhesion', AdSlot.HIDDEN_EVENT, () => {
+			porvataClosedActive = false;
+		});
```

Comment:
I would give it its own method, maybe even class, but I would have to see the bigger picture for that decision. But private method seems appropriate.

Reply:
Class is an overkill I guess, unless we think about bigger code-prio refactor. But private method is a good idea. Done.